packages/web-app-pdf-viewer/src/App.vue

```diff
@@ -36,8 +36,13 @@ export default defineComponent({
     url: '',
     resource: null
   }),
-  created() {
-    this.loadPdf(this.currentFileContext)
+  watch: {
+    currentFileContext: {
+      handler: function () {
+        this.loadPdf(this.currentFileContext)
```

Comment:
Please call `this.unloadPdf()` before the `loadPdf` call to free up unneeded blobs from the browser memory. Doesn't need any checks, `unloadPdf` does an early return if the url var is empty.